### PR TITLE
.github: create reuseable housekeeping workflow

### LINKED ISSUE #11: Housekeeping Workflow
# Overview

There are a number of manual tasks being performed by the team that can and should be automated. These include:

- Adding labels
- Adding Assignees
- Adding reviewers
- Adding to projects

# Help Actions

- Labels, see https://github.com/celestiaorg/celestia-node/issues/1259
- PR Template task checker https://github.com/marketplace/actions/task-completed-checker
- Add to projects https://github.com/marketplace/actions/add-to-github-projects
- Custom review assignees https://github.com/marketplace/actions/auto-request-review

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

This PR introduced a reuseable workflow with some common housekeeping actions, like adding labels and add issues and PRs to projects.

It also includes and example of how to call the workflow. 

I tested this on a personal repository prior to publishing here.

Team Leads and use this as a reference for their own repos if they would like to utilize some of the issue labeling and project management features.

`reuseable_housekeeping.yml` is the reusable workflow that can be called by each repo.

`housekeeping.yml` will be unique for each repo. The example created in this PR is specific to this repo. It can be used as a template for other repos and they can change how they are calling the reusable workflow. 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

Closes #11

### CODE REVIEW

File: .github/workflows/housekeeping.yml
Comment (liamsi) [inline .github/workflows/housekeeping.yml:10]: Question: Will that project board become the default for epics or will we still just collect all issues here?

Comment (MSevey) [reply .github/workflows/housekeeping.yml:10]: this was just for the example, each repo can define their own `housekeeping.yml` flow that determines their own project boards they want issues and/or PRs assigned to.

so there is no impact to other repos by defining this here. 

this workflow is showing how to add PRs with the `project` label to project number 10. 

You can even define multiple projects with a matrix to assign issues to multiple projects if a team has a repo specific project board and then a team level project board.

File: .github/auto_request_review.yml
Comment (tac0turtle) [inline .github/auto_request_review.yml:2]: out of curiosity what is the difference between this a code owners? 

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

Comment (MSevey) [reply .github/auto_request_review.yml:2]: the difference comes into play with the branch protection rules. 

one of the branch protection rules is that to require Code Owner approval, which is a good setting to have enabled.

However if we are using the code owners to auto assign PRs for the entire team (which is basically what we are doing now), it then makes the fact that we have code owners a moot point. 

This allows use to have specific people on the team review specific things, while maintain a select few code owners.

File: .github/workflows/housekeeping.yml
Comment (SweeXordious) [inline .github/workflows/housekeeping.yml:43]: [Question]
Is this only for Core-app? Or, the other repos, like node, also should have an EPIC board, etc

**Review by SweeXordious — COMMENTED** [review]
LGTM, thanks a lot for managing this stuff :fire: :fire: 
Left a non-blocking question.

File: .github/workflows/housekeeping.yml
Comment (MSevey) [reply .github/workflows/housekeeping.yml:43]: the `housekeeping.yml` action will be unique per repo. 

so each repo can pass in whatever project urls they want.

**Comment by Bidon15** [comment]
cc: @sysrex 

tagging Alex here as we need to think about how issues can be mentioned automatically in our board about some changes across different repos

**Review by evan-forbes — APPROVED** [review]
utACK

I still need to experiment with some these things or see them in action, but they seem really cool!!

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**
